### ISSUE TITLE
-Werror failure compiling with FreeBSD system clang 3.7.0

### ISSUE THREAD

**Issue by emaste** [davidchisnall/dtc#1]
e.g.
```
input_buffer.cc:228:6: error: 'precedence' overrides a member function but is
      not marked 'override' [-Werror,-Winconsistent-missing-override]
        int precedence()
            ^
input_buffer.cc:157:14: note: overridden virtual function is here
        virtual int precedence() = 0;
                    ^
input_buffer.cc:444:11: error: redundant move in return statement
      [-Werror,-Wredundant-move]
                        return std::move(lhs);
                               ^
input_buffer.cc:444:11: note: remove std::move call here
                        return std::move(lhs);
                               ^~~~~~~~~~   ~
```

**Comment by emaste** [comment]
There are still the `std::move` warnings

**Comment by davidchisnall** [comment]
This should be properly fixed by 15e6e6e
